examples/source_separation/utils/dist_utils.py

```diff
@@ -0,0 +1,74 @@
+import os
+import types
+import logging
+
+import torch
+import torch.distributed as dist
+
+
+def _info_on_master(self, *args, **kwargs):
+    if dist.get_rank() == 0:
+        self.info(*args, **kwargs)
+
+
+def getLogger(name):
+    """Get logging.Logger module with additional ``info_on_master`` method."""
+    logger = logging.getLogger(name)
+    logger.info_on_master = types.MethodType(_info_on_master, logger)
+    return logger
+
+
+_LG = getLogger(__name__)
+
+
+def setup_distributed(
+    world_size, rank, local_rank, backend="nccl", init_method="env://"
+):
+    """Perform env setup and initialization for distributed training"""
+    if init_method == "env://":
+        _set_env_vars(world_size, rank, local_rank)
+    params = {
+        "backend": backend,
+        "init_method": init_method,
+        "world_size": world_size,
+        "rank": rank,
+    }
+    _LG.info("Initializing distributed process group with %s", params)
+    dist.init_process_group(**params)
+    _LG.info("Initialized distributed process group.")
+
+
+def _set_env_vars(world_size, rank, local_rank):
+    for key, default in [("MASTER_ADDR", "127.0.0.1"), ("MASTER_PORT", "29500")]:
+        if key not in os.environ:
+            os.environ[key] = default
+
+    os.environ["WORLD_SIZE"] = str(world_size)
+    os.environ["RANK"] = str(rank)
+    os.environ["LOCAL_RANK"] = str(local_rank)
```

Comment:
If you were to let slurm spawn the processes for you (which can be handy in multi-node training) here is the recipe I  have that works fine on both slurm and with `torch.distributed.launch` https://github.com/facebookresearch/detr/blob/5e66b4cd15b2b182da347103dd16578d28b49d69/util/misc.py#L406-L428

Reply:
Yeah, I studied the same script from torchvision/reference/segmentation while I was studying the distributed training, and I came to the conclusion that not including SLURM-specific code makes the implementation simpler yet usable for SLARM. Check out the README bellow.

https://github.com/mthrok/audio/tree/conv-tasnet/examples/source_separation#slurm